torch/testing/_internal/common_utils.py

```diff
@@ -2060,8 +2060,9 @@ def genSparseTensor(self, size, sparse_dim, nnz, is_uncoalesced, device, dtype):
         i.mul_(torch.tensor(size[:sparse_dim]).unsqueeze(1).to(i))
         i = i.to(torch.long)
         if is_uncoalesced:
-            v = torch.cat([v, torch.randn_like(v)], 0)
-            i = torch.cat([i, i], 1)
+            i1 = i[:, :(nnz // 2), ...]
+            i2 = i[:, :(nnz // 2 + (0 if nnz % 2 == 0 else 1)), ...]
+            i = torch.cat([i1, i2], 1)
```

Comment:
This was silently creating `2 * nnz` elements with `is_coalesced=False`. Very unexpected for some of the corner cases tested in this PR.